### PR TITLE
VR-53 Adding VR Rotation digital camera rotations

### PR DESCRIPTION
**Pull request by SolarCrane**

### CODE REVIEW

File: CoreScriptsRoot/Modules/Settings/Pages/GameSettings.lua [hunk 1]
```diff
@@ -53,6 +54,7 @@ local utility = require(RobloxGui.Modules.Settings.Utility)
 RobloxGui:WaitForChild("Modules"):WaitForChild("TenFootInterface")
 RobloxGui:WaitForChild("Modules"):WaitForChild("Settings"):WaitForChild("SettingsHub")
 local isTenFootInterface = require(RobloxGui.Modules.TenFootInterface):IsEnabled()
+local HasVRAPI = pcall(function() return UserInputService.VREnabled and UserInputService.GetUserCFrame end)
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Settings/Pages/GameSettings.lua]: we are checking here for two things right? 1) if we have the api for vr and 2) if we have a vr device connected? Maybe I am remembering this wrong but isn't the first value just a bool whether the pcall succeeded?

File: CoreScriptsRoot/Modules/Settings/Pages/GameSettings.lua [hunk 2]
```diff
@@ -289,6 +291,31 @@ local function Initialize()
 			end)
 		end
 
+
+		------------------------------------------------------
+		------------------
+		------------------ VR Camera Mode -----------------------
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Settings/Pages/GameSettings.lua]: I think these options are all well and good, but product might want to see it

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx [hunk 3]
```diff
@@ -334,6 +375,64 @@ local function CreateCamera()
 		end
 	end
 
+	local functionExists = true
```
Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx]: seems like a bit of a generic name

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx [hunk 4]
```diff
@@ -334,6 +375,64 @@ local function CreateCamera()
...
+	function this:ShouldUseVRRotation()
+		if not functionExists then return false end
+		
+		local success, vrRotationIntensity = pcall(function() return StarterGui:GetCore("VRRotationIntensity") end)
+		if vrRotationIntensity == nil then
+			functionExists = false
+		end
+		return HasVRAPI and UserInputService.VREnabled and success and vrRotationIntensity ~= nil and vrRotationIntensity ~= "Smooth"
```
Comment (hockeychaos) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx]: we have to do a string compare every frame for this right?

File: CoreScriptsRoot/Modules/Settings/Pages/GameSettings.lua (hunk 2 above)
Comment (SolarCrane) [inline CoreScriptsRoot/Modules/Settings/Pages/GameSettings.lua]: Showed it to Tara, she is cool with it

**Pull request merged**